compiler/noirc_frontend/src/monomorphization/ast.rs

```diff
@@ -277,7 +277,7 @@ pub struct Function {
     pub unconstrained: bool,
     pub inline_type: InlineType,
     pub func_sig: FunctionSignature,
-    pub formal_verification_attributes: Vec<ResolvedFvAttribute>,
+    pub formal_verification_expressions: Vec<Expression>,
```

Comment:
The way you have implemented it we are losing information if the `Expression` is for an ensures attribute or a require attribute. You should create an enum like the ResolvedFvAttribute which contains ensure/requires and their expressions

Reply:
Also clear on line 10 the unused imports.